Source/Synchronization/Strategies/PushTokenStrategy.swift

```diff
@@ -148,12 +148,20 @@ extension PushTokenStrategy : ZMUpstreamTranscoder {
             guard let responseData = response.rawData else { return false }
             guard let payload = try? JSONDecoder().decode([String : [PushTokenPayload]].self, from: responseData) else { return false }
             guard let tokens = payload["tokens"] else { return false }
-
+            
             // Find tokens belonging to self client
-            let current = tokens.filter { $0.client == client.remoteIdentifier }
-
-            if current.count == 1 && // We found one token
-                current[0].token == pushToken.deviceTokenString // It matches what we have locally
+            let clientTokens = tokens.filter { $0.client == client.remoteIdentifier }
+            
+            let voipTokens = clientTokens.filter { $0.transport.contains(PushTokenType.voip.transportType) }
```

Comment:
Since you only care about the first VoIP token you could do:

```suggestion
            let voipToken = clientTokens.first(where: { $0.transport.contains(PushTokenType.voip.transportType) })
```